component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -126,15 +129,26 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             throws AuthenticationFailedException {
         try {
             String username = null;
+            String email = null;
+            boolean isEmailOTPMandatory;
             AuthenticatedUser authenticatedUser;
-            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
             Map<String, String> emailOTPParameters = getAuthenticatorConfig().getParameterMap();
             String tenantDomain = context.getTenantDomain();
             context.setProperty(EmailOTPAuthenticatorConstants.AUTHENTICATION, EmailOTPAuthenticatorConstants.AUTHENTICATOR_NAME);
             if (!tenantDomain.equals(EmailOTPAuthenticatorConstants.SUPER_TENANT)) {
                 IdentityHelperUtil.loadApplicationAuthenticationXMLFromRegistry(context, getName(), tenantDomain);
+                Object getPropertiesFromLocal = context.getProperty(IdentityHelperConstants.GET_PROPERTY_FROM_REGISTRY);
+                if (getPropertiesFromLocal == null) {
+                    isEmailOTPMandatory = Boolean.parseBoolean(String.valueOf(context.getProperty
+                            (EmailOTPAuthenticatorConstants.IS_EMAILOTP_MANDATORY)));
```

Comment:
In tenant when we load the xml file in registry, it will load the parameters in context, so we tried to get the parameters from context.